### ISSUE TITLE
failure in test_rcache.merge_expand_prot/mt-6 with valgrind

### ISSUE THREAD

**Issue by yosefe** [openucx/ucx#950]
http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1663/label=hpc-test-node/consoleFull

```
18:10:48 + taskset -c 8,9 timeout 80m make -C test/gtest UCS_HANDLE_ERRORS=bt 'VALGRIND_EXTRA_ARGS=--xml=yes --xml-file=valgrind.xml --child-silent-after-fork=yes' test_valgrind
...
18:11:01 [     INFO ] Using random seed of 23685
18:11:01 [     INFO ]  Setting for valgrind: UCX_IB_RX_QUEUE_LEN=512
18:11:01 [     INFO ]  Setting for valgrind: UCX_IB_RX_BUFS_GROW=512
18:11:01 [     INFO ]  Setting for valgrind: UCX_MM_RX_BUFS_GROW=32
18:11:01 [     INFO ]  Setting for valgrind: UCX_IB_TX_QUEUE_LEN=128
18:11:01 [     INFO ]  Setting for valgrind: UCX_IB_TX_BUFS_GROW=64
18:11:01 [     INFO ]  Setting for valgrind: UCX_RC_TX_CQ_LEN=256
18:11:01 [     INFO ]  Setting for valgrind: UCX_CM_TIMEOUT=600ms
18:11:02 Note: Randomizing tests' orders with a seed of 79613 .
18:11:02 [==========] Running 3026 tests from 402 test cases.
18:11:02 [----------] Global test environment set-up.
```

```
18:11:50 [       OK ] test_rcache.merge_merge_unwritable/mt-6 (248 ms)
18:11:50 [ RUN      ] test_rcache.merge_expand_prot/mt-6
18:11:50 /scrap/jenkins/scrap/workspace/hpc-ucx-pr/label/hpc-test-node/contrib/../test/gtest/ucs/test_rcache.cc:423: Failure
18:11:50 Expected: (region2->super.super.start) <= ((uintptr_t)ptr1), actual: 68096000 vs 68055040
18:11:50 /scrap/jenkins/scrap/workspace/hpc-ucx-pr/label/hpc-test-node/contrib/../test/gtest/ucs/test_rcache.cc:425: Failure
18:11:50 Value of: region2->super.prot
18:11:50   Actual: '\x2' (2)
18:11:50 Expected: 0x1|0x2
18:11:50 Which is: 3
18:11:51 
[  FAILED  ] test_rcache.merge_expand_prot/mt-6, where TypeParam =  and GetParam() =  (308 ms)
18:11:51 [ RUN      ] test_rcache.merge_with_unwritable/mt-6
18:11:51 [       OK ] test_rcache.merge_with_unwritable/mt-6 (153 ms)
```

**Comment by yosefe** [comment]
happened again in http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1733/label=hpc-test-node/consoleFull
```
22:04:34 [ RUN      ] test_rcache.merge_expand_prot/mt-6
22:04:34 /scrap/jenkins/scrap/workspace/hpc-ucx-pr/label/hpc-test-node/contrib/../test/gtest/ucs/test_rcache.cc:423: Failure
22:04:34 Expected: (region2->super.super.start) <= ((uintptr_t)ptr1), actual: 68317184 vs 68276224
22:04:34 /scrap/jenkins/scrap/workspace/hpc-ucx-pr/label/hpc-test-node/contrib/../test/gtest/ucs/test_rcache.cc:425: Failure
22:04:34 Value of: region2->super.prot
22:04:34   Actual: '\x2' (2)
22:04:34 Expected: 0x1|0x2
22:04:34 Which is: 3
22:04:34 
[  FAILED  ] test_rcache.merge_expand_prot/mt-6, where TypeParam =  and GetParam() =  (323 ms)
22:04:34 [ RUN      ] test_rcache.merge/mt-6
```

**Comment by yosefe** [comment]
fixed in #1088
